Sources/App/entrypoint.swift

```diff
@@ -1,21 +1,29 @@
 import Vapor
 import Logging
+import NIOCore
+import NIOPosix
 
 @main
 enum Entrypoint {
     static func main() async throws {
         var env = try Environment.detect()
         try LoggingSystem.bootstrap(from: &env)
         
-        let app = Application(env)
-        defer { app.shutdown() }
+        let app = try await Application.make(env)
+
+        // This attempts to install NIO as the Swift Concurrency global executor.
+        // You should not call any async functions before this point.
```

Comment:
@0xTim - we state here "You should not call any async functions before this point" but the template itself calls the async `Application.make(_:)` function. I'm assuming this is safe, but what are the limits?

It feels a bit weird to say "Don't do this" when the template itself does exactly that.

Reply:
Yeah fair. Essentially you'll hit assertion failures so it should be pretty obvious. #136 updates the comment and also disables it by default as it has caused some issues with shutdown so we'll make it opt in